library/core/src/iter/traits/iterator.rs

```diff
@@ -64,6 +67,7 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item = ()>) {}
 #[doc(notable_trait)]
 #[rustc_diagnostic_item = "Iterator"]
 #[must_use = "iterators are lazy and do nothing unless consumed"]
+#[cfg_attr(not(bootstrap), const_trait)]
```

Comment:
Something that was brought up in today's libs-api discussion:

Shouldn't there be something that marks this as unstably-const so that even if the const-trait lang feature is stabilized the library declaration that this a const trait is doesn't get stabilized? Like `rustc_const_unstable` on inherent impls.

Reply:
That would probably have to wait for the keyword_generics impl to make it easier to gate, which @oli-obk is currently working on.